src/odemis/driver/simulated.py

```diff
@@ -339,3 +238,195 @@ def __init__(self, name, role, positions, has_pressure=False, **kwargs):
+            if vachoices is None:
+                vachoices = {}
+            for n in vanames:
+                if n in vavalues:
+                    vavaluetype = vavalues[n].__class__.__name__.capitalize()
+                    if vavaluetype == "Bool":
+                        vavaluetype = "Boolean"
+                else:
+                    vavaluetype = "VA"
+                if n in varanges:
+                    varangetype = "Continuous"
+                elif n in vachoices:
+                    varangetype = "Enumerated"
+                else:
+                    varangetype = "VA"
+                if vavaluetype == "VA" and varangetype == "VA":
+                    vatype = "VigilantAttribute"
+                elif vavaluetype == "VA":  # and varangetype != "VA"
+                    vatype = varangetype + vavaluetype
+                else:
+                    vatype = vavaluetype + varangetype
+
+                vavalue = vavalues.get(n, 0.0)
+                ro = readonly.get(n, False)
+                vaunit = units.get(n, "")
+                varange = varanges.get(n, None)
+                vachoice = vachoices.get(n, None)
+                va_command = f"model.{vatype}({vavalue}, readonly={ro}, unit=\"{vaunit}\""
+                if varange:
+                    va_command += f", range={varange}"
+                elif vachoice:
+                    va_command += f", choice={vachoice}"
+                va_command += ")"
+                setattr(self, n, eval(va_command))
+
+        # Create desired axes
+        axes_def = {}
+        if axes:
+            if axisranges is None:
+                axisranges = {}
+            if axischoices is None:
+                axischoices = {}
+            self._position = {}
+            init_speed = {}
+            for a in axes:
+                init_speed[a] = 1.0  # we are fast!
+                rng = axisranges.get(a, None)
+                if rng is None:  # for this axis, no range was defined
+                    chs = axischoices.get(a, None)
+                    if chs is None:  # if no choice is defined either
+                        rng = (-0.1, 0.1)  # use the default range
+                    else:
+                        axes_def[a] = model.Axis(unit="m", choices=chs)
+                        self._position[a] = list(chs.values())[0]  # start at the first value
+                        continue
+                axes_def[a] = model.Axis(unit="m", range=rng, speed=(0., 10.))
+                self._position[a] = (rng[0] + rng[1]) / 2  # start at the centre
+
+            self._executor = model.CancellableThreadPoolExecutor(max_workers=1)
+
+            # RO, as to modify it the client must use .moveRel() or .moveAbs()
+            self.position = model.VigilantAttribute({}, unit="m", readonly=True)
+
+            self.speed = model.MultiSpeedVA(init_speed, (0., 10.), "m/s")
+
+        model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
+
```

Comment:
I attempted to delete the actuator specific methods and attributes here if no axes were defined, but this proved harder than expected.
Methods cannot be removed, since those are class specific, not instance specific. So they are either present for all instances or for none.
Attempting to delete attributes like self.axes, resulted in an AttributeError: cant' delete attribute.
I wasn't sure if deleting these attributes was worth diving deeper into, to track the origin of this exception and circumvent it somehow. (A quick google search yielded no result.)

Reply:
Can you not overwrite a method with a function which returns a "NotImplementedError" in case these methods are really annoying? Or overwrite the method with a variable containing None?

I also found this solution, which raises an error whenever the method/attribute foo is called.
```
    def __getattr__(self, attr_name):
        if attr_name == 'foo': raise AttributeError
        return getattr(self._wrapped, attr_name)
```
From: https://stackoverflow.com/questions/231839/python-inheritance-how-to-disable-a-function

However it seems to be all a non-pythonic solution. For a nice implementation you should probably create a different parent class which doesn't contain that method....